src/opendlv-statsd.cpp

```diff
@@ -47,27 +59,90 @@ int32_t main(int32_t argc, char **argv) {
 
             {
+                //retrieve latitude and longitude from the GeodeticWgs84Reading
+                double latitude=gps.latitude();
+                double longitude=gps.longitude();   
+                
+                if(VERBOSE) 
+                    std::cerr << "Got " << latitude << ", " << longitude << " as Envelopes" <<std::endl;
+                
+                std::lock_guard<std::mutex> lck(dataMutex);
+                
+                //push the data into queues
+                lat_queue.insert(lat_queue.end(),latitude);
+                lon_queue.insert(lon_queue.end(),longitude);
+
                 // TODO: Temporarily store the desired field to be passed on to statsd in the time-trigger below.
             }
         };
-
+        
+        
         // Register our lambda for the message identifier for opendlv::proxy::GeodeticWgs84Reading.
         od4.dataTrigger(opendlv::proxy::GeodeticWgs84Reading::ID(), onGPS);
-
+        
         // Time-triggered sender; TODO: reflect the desired sample rate via commandline.
-        const float ONE_HZ{1.0f};
-        od4.timeTrigger(ONE_HZ, [&od4, &dataMutex](){
+        float FREQUENCY{std::stof(sample_rate+"f")};
+        od4.timeTrigger(FREQUENCY, [&od4, &dataMutex, &lat_queue, &lon_queue ,VERBOSE](){
             {
                 std::lock_guard<std::mutex> lck(dataMutex);
-                // TODO: something with the WGS84 reading, i.e. forward to statsd.
+                
+                std::stringstream ss;
+                std::string packet;
+                
+                //the following loop will optimize the packet size to 1462 (or other specified)
+                do{
+                    double lat=0,lon=0;
+
+                    //store first items in queues 
+                    if(!lat_queue.empty()){
+                        lat = lat_queue.back();
+                    }
+
+
+                    if(!lon_queue.empty()){
+                        lon = lon_queue.back();
+                    }
+                    
+                    //pack the stored items in statsd format
+                    ss<<packet<<"la:"<<lat<<"|c"<<"\n"<<"lo:"<<lon<<"|c";
+                    
+                    if(ss.str().size()<=MAX_PACKET_SIZE){
+                        if(!lat_queue.empty())
+                            lat_queue.erase(lat_queue.end()-1);
+                        if(!lon_queue.empty())
+                            lon_queue.erase(lon_queue.end()-1);
+                        
+                        ss<<"\n";
+                        packet = ss.str();
+                        ss.str("");
+                    }
+                                     
+                }while((packet.size() + ss.str().size())<=MAX_PACKET_SIZE-1);
+                
+                
+                
+                //create udp sender to statsd
+                cluon::UDPSender s("127.0.0.1", 8125);
```

Comment:
The `statsd` IP and port should be optional command line parameters, defaulting to `127.0.0.1` and `8125`.